### PR TITLE
C_BPartner_QuickInput.MKTG_Campaign_ID -> C_BPartner.MKTG_Campaign_ID

### PR DESCRIPTION
**Pull request by teosarca**

### CODE REVIEW

File: backend/de.metas.business/src/main/java/de/metas/bpartner/composite/BPartner.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import de.metas.bpartner.BPGroupId;
 import de.metas.bpartner.BPartnerId;
 import de.metas.bpartner.OrgMappingId;
+import de.metas.campaign.CampaignId;
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/bpartner/composite/BPartner.java:7]: use de.metas.marketing.base.model.CampaignId

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/campaign/CampaignId.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import com.fasterxml.jackson.annotation.JsonValue;
+import de.metas.util.Check;
+import de.metas.util.lang.RepoIdAware;
+import lombok.Value;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+@Value
+public class CampaignId implements RepoIdAware
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/campaign/CampaignId.java:35]: drop this class, it won't be used

File: backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositeSaver.java [hunk 3]
```diff
@@ -17,6 +17,7 @@
 import de.metas.bpartner.composite.BPartnerLocationAddressPart;
 import de.metas.bpartner.composite.BPartnerLocationType;
 import de.metas.bpartner.service.IBPartnerBL;
+import de.metas.campaign.CampaignId;
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositeSaver.java:20]: use de.metas.marketing.base.model.CampaignId

File: backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositesLoader.java [hunk 4]
```diff
@@ -22,6 +22,7 @@
 import de.metas.bpartner.composite.BPartnerLocation;
 import de.metas.bpartner.composite.BPartnerLocationAddressPart;
 import de.metas.bpartner.composite.BPartnerLocationType;
+import de.metas.campaign.CampaignId;
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositesLoader.java:25]: use de.metas.marketing.base.model.CampaignId

File: backend/de.metas.business/src/main/java/de/metas/bpartner/quick_input/service/BPartnerQuickInputService.java [hunk 5]
```diff
@@ -41,6 +41,7 @@
 import de.metas.bpartner.quick_input.BPartnerContactQuickInputId;
 import de.metas.bpartner.quick_input.BPartnerQuickInputId;
 import de.metas.bpartner.service.IBPGroupDAO;
+import de.metas.campaign.CampaignId;
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/bpartner/quick_input/service/BPartnerQuickInputService.java:44]: use de.metas.marketing.base.model.CampaignId

**Review by teosarca — COMMENTED** [review]
check my comments

File: backend/de.metas.business/src/main/java/de/metas/bpartner/composite/BPartner.java (hunk 1 above)
Comment (BodorAnnamaria) [reply backend/de.metas.business/src/main/java/de/metas/bpartner/composite/BPartner.java]: Changed

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/campaign/CampaignId.java (hunk 2 above)
Comment (BodorAnnamaria) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/campaign/CampaignId.java]: deleted

File: backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositeSaver.java (hunk 3 above)
Comment (BodorAnnamaria) [reply backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositeSaver.java]: changed

File: backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositesLoader.java (hunk 4 above)
Comment (BodorAnnamaria) [reply backend/de.metas.business/src/main/java/de/metas/bpartner/composite/repository/BPartnerCompositesLoader.java]: changed

File: backend/de.metas.business/src/main/java/de/metas/bpartner/quick_input/service/BPartnerQuickInputService.java (hunk 5 above)
Comment (BodorAnnamaria) [reply backend/de.metas.business/src/main/java/de/metas/bpartner/quick_input/service/BPartnerQuickInputService.java]: changed

**Pull request merged**